src/devices/Common/System/Device/Analog/AnalogController.cs

```diff
@@ -0,0 +1,153 @@
+        public abstract bool SupportsAnalogInput(int pin);
+
+        /// <summary>
+        /// Convert the input pin number to logical pin numbers.
+        /// </summary>
+        /// <param name="pinNumber">Pin Number, in the numbering scheme of the analog controller</param>
+        /// <returns>Logical pin number</returns>
+        public virtual int ConvertPinNumberToLogicalNumberingScheme(int pinNumber)
+        {
+            return pinNumber;
+        }
+
+        /// <summary>
+        /// Convert logical pin to caller's pin numbering scheme.
+        /// </summary>
+        /// <param name="logicalPinNumber">Logical pin numbering of the board</param>
+        /// <returns>Pin number in the scheme of the parent analog controller</returns>
+        public virtual int ConvertLogicalNumberingSchemeToPinNumber(int logicalPinNumber)
+        {
+            return logicalPinNumber;
+        }
+
+        /// <summary>
+        /// Opens a pin in order for it to be ready to use.
+        /// </summary>
+        /// <param name="pinNumber">The pin number in the controller's numbering scheme.</param>
+        public AnalogInputPin OpenPin(int pinNumber)
+        {
+            // This does the number conversion itself, therefore done first
+            if (!SupportsAnalogInput(pinNumber))
+            {
+                throw new NotSupportedException($"Pin {pinNumber} is not supporting analog input.");
+            }
+
+            int logicalPinNumber = ConvertPinNumberToLogicalNumberingScheme(pinNumber);
+
+            if (_openPins.Any(x => x.PinNumber == pinNumber))
+            {
+                throw new InvalidOperationException("The selected pin is already open.");
+            }
+
+            AnalogInputPin openPin = OpenPinCore(logicalPinNumber);
+            _openPins.Add(openPin);
+            return openPin;
+        }
+
+        /// <summary>
+        /// Overriden by derived classes: Returns an instance of the <see cref="AnalogInputPin"/> for the specified pin.
+        /// </summary>
+        /// <param name="pinNumber">Pin number, in the logical pin numbering scheme</param>
+        /// <returns>An instance of a pin</returns>
+        protected abstract AnalogInputPin OpenPinCore(int pinNumber);
+
+        /// <summary>
+        /// Checks if a specific pin is open.
+        /// </summary>
+        /// <param name="pinNumber">The pin number in the controller's numbering scheme.</param>
+        /// <returns>The status if the pin is open or closed.</returns>
+        public virtual bool IsPinOpen(int pinNumber)
+        {
+            int logicalPinNumber = ConvertPinNumberToLogicalNumberingScheme(pinNumber);
+            return _openPins.Any(x => x.PinNumber == logicalPinNumber);
+        }
+
+        /// <summary>
+        /// Closes the given pin
+        /// </summary>
+        /// <param name="pin">The pin to close</param>
+        public virtual void Close(AnalogInputPin pin)
```

Comment:
ClosePin/CloseAnalogPin - make sure to match the name with OpenPin/OpenAnalogPin

Reply:
Yea, calling it `ClosePin`.